Don't modify the where clause on empty inputs.

markstory opened the pull request:
When we get '' or [] we can safely assume that no where clause should be set, as it would result in an invalid query.

Refs #7028

markstory committed:
Don't modify the where clause on empty inputs.

When we get '' or [] we can safely assume that no where clause should be
set, as it would result in an invalid query.

Refs #7028

lorenzo commented on src/Database/Query.php:
```diff
@@ -755,6 +755,9 @@ protected function _makeJoin($table, $conditions, $type)
     {
+        if (empty($conditions)) {
+            return $this;
+        }
```
I think this should be put in the `_conjugate` function

markstory replied:
Good idea. Fixed now.

markstory committed:
Move empty check to a more general purpose place.

empty values generally misbehave in all the various parts of query
building so we'll omit them.
